src/components/ContactPreview.tsx

```diff
@@ -0,0 +1,50 @@
+'use client'
+
+import { css } from '../../styled-system/css'
+import useAvatar from '@/components/useAvatar'
+import { useRouter } from 'next/navigation'
+import { createClientComponentClient } from '@supabase/auth-helpers-nextjs'
+import { Database } from '@/types/supabase'
+import { useContext } from 'react'
+import { UserContext } from '@/components/UserProvider'
+import { Profile } from '@/types/models'
+
+type Props = {
+  contact: Profile
+}
+
+export default function ContactPreview({ contact }: Props) {
+  const supabase = createClientComponentClient<Database>()
+  const router = useRouter()
+  const { user } = useContext(UserContext)
+
+  const { AvatarPreview } = useAvatar({
+    avatarUrl: contact.avatar_url,
+  })
+
+  const handleClick = async () => {
+    // TODO: check the chatId not to create multiple chats for the same users
```

Comment:
bug: Yeah... I'm struggling to test a 1-1 chat because:
- I don't know which accounts are mine because I can't update the username (as reported above)
- Each time I click a user it creates a new chat instead of reusing the existing one.

I could do it by checking the logs, DB and hardcode the URL but that's far from ideal 😅